package com.oguzhansecgel.to_do_app.exception;

import java.util.Date;
import java.util.Map;

public class ApiResult {
    private int status;
    private String error;
    private String message;
    private String path;
    private Map<String,Object> validationErrors;
    private Date createdDate = new Date(System.currentTimeMillis());

    public ApiResult() {
    }

    public ApiResult(int status, String error, String message, String path, Map<String, Object> validationErrors, Date createdDate) {
        this.status = status;
        this.error = error;
        this.message = message;
        this.path = path;
        this.validationErrors = validationErrors;
        this.createdDate = createdDate;
    }

    public int getStatus() {
        return status;
    }

    public void setStatus(int status) {
        this.status = status;
    }

    public String getError() {
        return error;
    }

    public void setError(String error) {
        this.error = error;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public String getPath() {
        return path;
    }

    public void setPath(String path) {
        this.path = path;
    }

    public Map<String, Object> getValidationErrors() {
        return validationErrors;
    }

    public void setValidationErrors(Map<String, Object> validationErrors) {
        this.validationErrors = validationErrors;
    }

    public Date getCreatedDate() {
        return createdDate;
    }

    public void setCreatedDate(Date createdDate) {
        this.createdDate = createdDate;
    }
}
